src/pins/pin.rs

```diff
@@ -238,7 +252,7 @@ where
         self,
         token: Token<T, init_state::Enabled>,
     ) -> GpioPin<T, direction::Input> {
-        GpioPin::new(token, ())
+        GpioPin::new(*token.pin_info(), ())
```

Comment:
With my suggested change above, we can no longer get this from `Token`, but I don't think we need to:
```suggestion
        GpioPin::new(self.ty, ())
```